### ISSUE TITLE
array parameter in_alias created globally in function _zsh_highlight_main_highlighter_highlight_list

### ISSUE THREAD

**Issue by jasonxia23** [zsh-users/zsh-syntax-highlighting#795]
Following warning will show out, while inputting ``for branch in `gb`; do [[ $branch != 'master' ]] && gbD $branch; done`` inside terminal:

```
_zsh_highlight_main_highlighter_highlight_list:228: array parameter in_alias created globally in function _zsh_highlight_main_highlighter_highlight_list
```

**Comment by phy1729** [comment]
There' isn't enough information to reproduce the error. Please provide a minimal reproducer starting from zsh -f and sourcing zsh-syntax-highlighting. I'm not sure how this could happen as one of the first things _zsh_highlight_main_highlighter_highlight_list does is local in_alias.

**Comment by jasonxia23** [comment]
I felt confused either, but had no idea about the warning. To reproduce this issue, just alias `gb` and `gbD` as below:

```sh
alias gb="git branch"
alias gbD="git branch -D"
```

and then copy ``for branch in `gb`; do [[ $branch != 'master' ]] && gbD $branch; done`` into terminal.

**Comment by danielshahaf** [comment]
I can't reproduce the bug in `zsh -f` with those aliases, under `zsh-5.8-286-ge404c23`.

Please post the output of `typeset -p ZSH_VERSION ZSH_PATCHLEVEL ZSH_HIGHLIGHT_REVISION ZSH_HIGHLIGHT_VERSION`.

@phy1729 I suspect a zsh bug that has been fixed in your zsh but not in @jasonxia23's.

**Comment by jasonxia23** [comment]
```sh
$ typeset -p ZSH_VERSION ZSH_PATCHLEVEL ZSH_HIGHLIGHT_REVISION ZSH_HIGHLIGHT_VERSION
typeset ZSH_VERSION=5.7.1
typeset ZSH_PATCHLEVEL=zsh-5.7.1-0-g8b89d0d
typeset ZSH_HIGHLIGHT_REVISION=HEAD
typeset ZSH_HIGHLIGHT_VERSION=0.8.0-alpha2-dev
```

**Comment by danielshahaf** [comment]
I can reproduce the warning with zsh 5.7.1 from Debian but not with my self-compiled `zsh-5.8-286-ge404c23` using exactly the same steps in both cases.

I don't see anything relevant in `git log Src` between these.  Anyone wants to bisect?  Or shall we just close this as an upstream bug?

**Comment by jasonxia23** [comment]
Have fixed this issue by upgrade zsh to 5.8. Thanks a lot for your replies.
